queries.sql

```diff
@@ -132,4 +132,140 @@ FROM animals an
 INNER JOIN owners ow
 ON ow.id = an.owners_id 
 GROUP BY ow.full_name 
-ORDER BY MAX(an.name) DESC;
\ No newline at end of file
+ORDER BY MAX(an.name) DESC;
+
+-- Who was the last animal seen by William Tatcher? 
+SELECT 
+an.name as animal_name,
+ve.name as vet_name,
+vi.visit_date
+FROM animals an
+INNER JOIN visits vi
+ON an.id = vi.animal_id
+INNER JOIN vets ve
+ON ve.id = vi.vet_id
+WHERE ve.name = 'William Tatcher' 
+ORDER BY vi.visit_date DESC LIMIT 1; 
+
+-- How many different animals did Stephanie Mendez see?
+SELECT 
+ve.name AS vet_name,
+COUNT(an.name) AS animal_count
+FROM
+animals an
+INNER JOIN visits vi
+ON an.id = vi.animal_id
+INNER JOIN vets ve
+ON ve.id = vi.vet_id
+WHERE ve.name = 'Stephanie Mendez'
+GROUP BY ve.name;
+
+-- List all vets and their specialties, including vets with no specialties.
+SELECT 
+DISTINCT 
+ve.name AS vet_name,
+sp.name AS animal_type 
+FROM
+vets ve
+LEFT JOIN specializations spe
+on ve.id = spe.vet_id 
+LEFT JOIN animals an 
+on spe.species_id = an.species_id 
+LEFT JOIN species sp
+ON  an.species_id = sp.id 
+ORDER BY ve.name;
```

Comment:
- [ ] [optional] It would be nice having a query that lists the vets without repetition of the vets' names.

`Maybe consider this`

```suggestion
    SELECT vets.*, species.name
  FROM vets
  LEFT JOIN specializations ON vets.id = specializations.vet_id
  LEFT JOIN species ON specializations.species_id = species.id;

```